### ISSUE TITLE
Backdrop z-Index forced to 1050

### ISSUE THREAD

**Issue by alexnoise79** [ng-bootstrap/ng-bootstrap#2686]
### Bug description:

Backdrop z-Index has been forced to 1050 with inline style

```html
<ngb-modal-backdrop style="z-index: 1050" class="modal-backdrop fade show">
```

this works if you use normal bootstrap.css, but i'm using bootstrap/scss and i like to configure myself the variables

in particular $zindex-modal: on my project is set much lower than 1050, and all the modals are broken with the updates.

if for some reason is not possible to change this behaviour, at least give us the possibility to configure it trough NgbModalConfig.

Thanks

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular:  6.1.3

ng-bootstrap: 3.2.0

Bootstrap: 4.1.3

**Comment by pkozlowski-opensource** [comment]
Thnx for reporting - this is a regression / side effect of 2409572aa1d6cfcbecaddf1c6bed68a43c366d47

@maxkorz do you recall why it was necessary to force z-index here: https://github.com/ng-bootstrap/ng-bootstrap/commit/2409572aa1d6cfcbecaddf1c6bed68a43c366d47#diff-5cffcc19b3f0a215c7be0ea19a593837R7 ?

**Comment by maxkorz** [comment]
@pkozlowski-opensource yes, z-index is required to make modals appear stacked.

With forced z-index: ![image](https://user-images.githubusercontent.com/25089646/45253058-66a7a180-b369-11e8-8de5-3e082ccb4451.png)
 Without forced z-index: ![image](https://user-images.githubusercontent.com/25089646/45253054-568fc200-b369-11e8-84e3-345aabcaa7b5.png)

I'll make a PR shortly

**Comment by maxokorokov** [comment]
So, basically it's because of the DOM nodes order:

![screen shot 2018-09-11 at 14 48 31](https://user-images.githubusercontent.com/8074436/45361225-010a1e00-b5d2-11e8-984e-a07961e55411.png)

Ideally we should have a SASS theme that would reuse the Bootstrap's variable for z-index, but for now allowing to customize it seems like a solution.

**Comment by ojemuyiwa** [comment]
<img width="1938" alt="screenshot 2018-12-18 at 13 35 42" src="https://user-images.githubusercontent.com/13540819/50157631-db27a000-02c9-11e9-8748-64fe15534a8c.png">
Stacking seems to work as well with inline styles removed and modal back-drop set to 1050

**Comment by agenaille** [comment]
Just wanted to point out a workaround. 
You can use the backdropClass to apply a css class to the ngb-modal-backdrop and then use the CSS !important to force override this.

```
.your-backdrop-class {
  z-index: 1060!important;
}
```

**Comment by alexnoise79** [comment]
@agenaille yes if you want you can force with an !important, and this is even what i have done in my override of variables.scss

```
$zindex-dropdown: 100;
$zindex-sticky: 102;
$zindex-fixed: 103;
$zindex-modal-backdrop: 104 !important; // fix for ng-bootstrap 3.2.0 todo: remove if fixed
$zindex-modal: 105;

```
but honestly !important is never a good solution for me

**Comment by majora2007** [comment]
Just wanted to say that by default, with the z-index 1055, when using an ngbDropdown within a ngbModal, by default due to the z-index, it will not render correctly (on v12).

**Comment by dhabierre** [comment]
I have an issue with v13 too.
The `z-index` is set to `1055` and the overlay is over all the page:

![image](https://user-images.githubusercontent.com/29678854/156932779-ec20d122-6da3-4d04-b133-0d28e6b32817.png)

HTML render:

```html
<ngb-modal-backdrop style="z-index: 1055;" aria-hidden="true" class="modal-backdrop fade show"></ngb-modal-backdrop>
```

If I change this value with 1050 (using DevTools) it is OK.
 
I can't see any possibility to override this value by code (css, backdropClass... don't work).

**modal-backdrop.ts**

```js
@Component({
  selector: 'ngb-modal-backdrop',
  encapsulation: ViewEncapsulation.None,
  template: '',
  host: {
    '[class]': '"modal-backdrop" + (backdropClass ? " " + backdropClass : "")',
    '[class.show]': '!animation',
    '[class.fade]': 'animation',
    'style': 'z-index: 1055' // <======== HERE
  }
})
export class NgbModalBackdrop implements OnInit {
```

Thanks.

**Comment by ojemuyiwa** [comment]
> I have an issue with v13 too. The `z-index` is set to `1055` and the overlay is over all the page:
> 
> ![image](https://user-images.githubusercontent.com/29678854/156932779-ec20d122-6da3-4d04-b133-0d28e6b32817.png)
> 
> HTML render:
> 
> ```
> <ngb-modal-backdrop style="z-index: 1055;" aria-hidden="true" class="modal-backdrop fade show"></ngb-modal-backdrop>
> ```
> 
> If I change this value with 1050 (using DevTools) it is OK.
> 
> I can't see any possibility to override this value by code (css, backdropClass... don't work).
> 
> **[modal-backdrop.ts](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/modal/modal-backdrop.ts)**
> 
> ```js
> @Component({
>   selector: 'ngb-modal-backdrop',
>   encapsulation: ViewEncapsulation.None,
>   template: '',
>   host: {
>     '[class]': '"modal-backdrop" + (backdropClass ? " " + backdropClass : "")',
>     '[class.show]': '!animation',
>     '[class.fade]': 'animation',
>     'style': 'z-index: 1055' // <======== HERE
>   }
> })
> export class NgbModalBackdrop implements OnInit {
> ```
> 
> Thanks.

*Maybe: Use a higher specificity in your css e.g. modal-backdrop.show

**Comment by ruifelizardo** [comment]
Ive got the same problem, backdrop on top of modal (and im using a dropdown also)

**Comment by dhabierre** [comment]
I retry following [this tutorial](https://www.tutsmake.com/angular-13-bootstrap-modal-popup-example/) and it is OK now.

**Comment by ruch0401** [comment]
`::ng-deep` will apply the style to all the child components. This solved the issue for me.

`
::ng-deep ngb-modal-backdrop {
  z-index: 1050 !important;
}
`

**Comment by unnatRevo** [comment]
> `::ng-deep` will apply the style to all the child components. This solved the issue for me. This does in your css file.
> 
> ```
> ::ng-deep ngb-modal-backdrop {
>   z-index: 1050 !important;
> }
> ```

This solved my issue. Thank you

**Comment by harikvpy** [comment]
+1

**Comment by friedrichroell** [comment]
> `::ng-deep` will apply the style to all the child components. This solved the issue for me. This does in your css file.
> 
> ```
> ::ng-deep ngb-modal-backdrop {
>   z-index: 1050 !important;
> }
> ```

actually adding the style to my global styles without ng-deep did the trick for me

```
ngb-modal-backdrop {
    z-index: 1050 !important;
}

```

**Comment by musmanfaisal** [comment]
> > `::ng-deep` will apply the style to all the child components. This solved the issue for me. This does in your css file.
> > ```
> > ::ng-deep ngb-modal-backdrop {
> >   z-index: 1050 !important;
> > }
> > ```
> 
> This solved my issue. Thank you

Thank you
Solved my issue

**Comment by panyann** [comment]
Maybe someone could just fix the issue, instead of us making stupid workarounds? The modals are still behind the backdrop
<img width="423" alt="image" src="https://user-images.githubusercontent.com/22815508/216559966-27177c14-84a3-4529-87ae-0da63c28a078.png">

**Comment by JoeMeeks** [comment]
> I've retried following [this tutorial](https://www.tutsmake.com/angular-13-bootstrap-modal-popup-example/) and it is OK now. I think I was missing this:
> 
> **angular.json**
> 
> ```css
> ...
> "styles": [
>   "node_modules/bootstrap/dist/css/bootstrap.min.css", <== this line
>   "src/styles.css"
> ],
> ...
> ```

Although the tutorial link is no longer served, this `boostrap.min.css` style reference fixed the issue in my ng17 projects.

**Comment by mcvendrell** [comment]
Same backdrop issue in ngbootstrap v17 (with Angular v18). Adding bootstrap.min.css (v4.6) didn't fix it.

I applied the solution by @friedrichroell and it's working:

```
ngb-modal-backdrop {
    z-index: 1050 !important;
}
```

**Comment by DerZyklop** [comment]
> ```
> ngb-modal-backdrop {
>     z-index: 1050 !important;
> }
> ```

This suggested workaround sets all modal-backdrops to appear behind all modals, no matter how many Modals you have. So it works fine if you have only one modal, but it breaks differently if you have stacked modals. 

The result would be this:

![Screenshot 2024-09-25 at 11 18 26](https://github.com/user-attachments/assets/69ad7d5c-e1a3-4aaf-a090-d35a8761c4b4)

Which is wrong because my warning modal should be covered by another backdrop.

**Comment by DerZyklop** [comment]
### Solution

Here is the solution that should work for all of you, no matter if you have one or multiple, stacked modals:

```
ngb-modal-window:last-of-type {
    z-index: 1056 !important;
}
```

### Explanation
 
The selector `ngb-modal-window:last-of-type` picks only the very last modal which has been appended to the DOM and sets it to be at the forefront.

### Result 

![Screenshot 2024-09-25 at 11 30 23](https://github.com/user-attachments/assets/44a06b91-eaf3-4c65-875b-df2b142fdec7)

**Comment by zkirill** [comment]
In my case I had to add the following to @DerZyklop's solution to make it work.

`ng-bootstrap ^12.1.2, bootstrap ~5.0.2, angular ^13.3.12`

```css
ngb-modal-backdrop {
  z-index: 1050 !important;
}

ngb-modal-window {
  z-index: 1050 !important;
}
```
